qemu/tests/virtio_trace_pipenb.py

```diff
@@ -0,0 +1,48 @@
+import time
+import os
+import errno
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Under named-pipe non-blocking testing:
+    1) Create pipe named by the following
+    2) Boot up a single-CPU guest with a virtio-serial device and
+       named-pipe chardev backend
+    3) Write data to the virtio-serial port until the guest stops.
+    4) check whether guest can work.
+    5) Read the named-pipe file on the host.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    timeout = float(params.get("login_timeout", 360))
+    vm = env.get_vm(params["main_vm"])
+    serials = params["serials"].split()
+    v_path = vm.get_serial_console_filename(serials[-1])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    out_put = session.cmd_output("nohup cat /proc/kallsyms > /dev/virtio-ports/vs2 2>&1 &")
+    time.sleep(10)
+    if session.cmd_status_output("date")[1] is None:
+        test.fail("There should be date in the output!")
```

Comment:
Better explain no date means guest freeze after write data to the virtio-serial port while there's no reader process in the host.

Reply:
update, please help to check.